pandas/core/generic.py

```diff
@@ -9094,9 +9093,12 @@ def mask(
         cond = com.apply_if_callable(cond, self)
 
         # see gh-21891
-        if not hasattr(cond, "__invert__"):
+        if not hasattr(cond, "__array__"):
```

Comment:
what needed this?

Reply:
This was to make sure we have an array so we can index as below (e.g., it's possible we get here with a scalar and the block below breaks, for example)